src/pages/crm-products-page/CrmProductsPage.tsx

```diff
@@ -2,28 +2,204 @@ import { Link } from 'react-router-dom'
 import PlusIcon from '../../icons/plus.svg?react'
 import SearchIcon from '../../icons/search.svg?react'
 import styles from './crmProductsPage.module.scss'
+import { GridPaginationModel } from '@mui/x-data-grid'
+import { SelectChangeEvent } from '@mui/material'
+import { ChangeEvent, useCallback, useEffect, useState } from 'react'
+import {
+  fetchMainCategories,
+  fetchProductsForCrm,
+  searchProductsForCrm
+} from '../../redux/crm-add-new-product/operation'
+import { useDispatch, useSelector } from 'react-redux'
+import { AppDispatch, RootState } from '../../redux/store'
+import { ProductStatus } from '../../types'
+import SelectProductStatus from './SelectProductStatus'
+import SelectProductCategory from './SelectProductCategory'
+import CustomPagination from './CrmProductPageCustomPagination'
+import { StyledDataGrid } from './CrmProductPageStyledDataGrid'
+import { columns } from './CrmProductDataGridColumns'
+import useDebounce from '../../hooks/useDebounce'
+
+export const PAGE_SIZE = 10
+
+export const OPTIONS = [
+  { value: '', label: 'Статус' },
+  { value: ProductStatus.NEW, label: 'Новий' },
+  { value: ProductStatus.ACTIVATED, label: 'Активний' },
+  { value: ProductStatus.ARCHIVED, label: 'Архіований' }
+]
 
 const CrmProductsPage = () => {
+  const [status, setStatus] = useState('')
+  const [category, setCategory] = useState('')
+  const [searchQuery, setSearchQuery] = useState('')
+  const searchDebounce = useDebounce(searchQuery, 300)
+  const dispatch = useDispatch<AppDispatch>()
+  const {
+    productsForCrmTotalCount,
+    productsForCrmPageNumber,
+    productsForCrm,
+    isLoadingForCrm,
+    searchProductForCrm,
+    searchProductsForCrmTotalCount,
+    searchProductsForCrmPageNumber,
+    isLoadingForCrmSearch
+  } = useSelector((state: RootState) => state.newProduct)
+
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({
+    pageSize: PAGE_SIZE,
+    page: searchQuery
+      ? searchProductsForCrmPageNumber
+      : productsForCrmPageNumber
+  })
+  const mainCategories = useSelector(
+    (state: RootState) => state.allCategories.mainCategories
+  )
+  const handleCategoryChange = (event: SelectChangeEvent) => {
+    setCategory(event.target.value)
+  }
+  const handleStatusChange = (event: SelectChangeEvent) => {
+    setStatus(event.target.value)
+  }
+
+  const handlePaginationModelChange = useCallback(
```

Comment:
why do you need `useCallback` here?

Reply:
I'm using it for better performance and I have dependencies [status, productsForCrmPageNumber, category, searchDebounce]

Reply:
this is good, but you should use `useCallback` and also `useMemo` when you really have performance issues and want to optimize something. If this is just a regular component, then using these optimizations will actually make your code worse. React itself can handle renders correctly

Reply:
callback as a value can fix some bugs, but there is a 5% chance that you need this type of setting state. In most of cases regular set state, without callback, can handle all of your needs